tests/unit/linalg/test_matrix_rectangular.cpp

```diff
@@ -27,7 +27,7 @@ void gradf1(const Vector &x, Vector &u)
 {
    u(0) = 2*x(0);
    if (x.Size() >= 2) { u(1) = 3*pow(x(1), 2); }
-   if (x.Size() >= 3) { u(2) = 4*pow(x(1), 3); }
+   if (x.Size() >= 3) { u(2) = 4*pow(x(2), 3); }
```

Comment:
I guess this didn't make a difference because these tests are only run on 2D meshes. Does it make sense to run a 3D test, too? It's weird to have code paths that are never run. cc: @jandrej